### ISSUE TITLE
+:emoji: reactions are added to (hidden) system messages

### ISSUE THREAD

**Issue by hloeffler** [mattermost/mattermost-server#7703]
#### Summary
 using: `+:emoji:` directly after a system message, will add the reaction to the system message, even if they are hidden (disabled via the options).

#### Steps to reproduce
* hide system messages
* wait for a system message (join/leave)
* use `+:emoji:`
 
#### Expected behaviour
Emoji reaction applies to last visible message

#### Observed behaviour (that appears unintentional) 
I see nothing, others see Emoji on system message (which is invisible for me)

#### Possible fixes
* Apply response to last visible message
* Disable emote responses for system messages

**Comment by lindy65** [comment]
Hi @hloeffler,

Thank you for your report!

We had [this ticket](https://mattermost.atlassian.net/browse/PLT-6220) which was fixed in v3.8.0 in April so emoji reactions should not be posted to system messages whether they're hidden or not.

That being said, I have reproduced your issue on our nightly build server running of master and have created [this ticket](https://mattermost.atlassian.net/browse/PLT-7974) to track.
